.gitignore

```diff
@@ -65,6 +65,12 @@ __pycache__/
 /provenance_repository
 /logs
 msi/WixWin.wsi
+extensions/expression-language/el-scanner/Scanner.cpp
+extensions/expression-language/el-parser/Parser.cpp
+extensions/expression-language/el-parser/Parser.hpp
+extensions/expression-language/el-parser/location.hh
+extensions/expression-language/el-parser/position.hh
+extensions/expression-language/el-parser/stack.hh
```

Comment:
Why did you extract the local gitignore to this root one?
I prefered the previous approach, because it was more modular.

Reply:
In the first draft el-scanner and el-parser directories were not part of the repository as they were only created at build time with the generated files. As CPPLINT.cfg is now also added in those directories I think you are right, I'll add the .gitignore files there as well.

Reply:
Updated in be9cace2da1aa161129ac9c22986033dafc199db